test/Source/DoctrineORMQuerySourceIteratorTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Exporter\Test\Source;
+
+use Doctrine\ORM\Configuration;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Query;
+use Exporter\Test\Source\Fixtures\DoctrineORMQuerySourceIterator;
+use Exporter\Test\Source\Fixtures\ObjectWithToString;
+
+
+class DoctrineORMQuerySourceIteratorTest extends \PHPUnit_Framework_TestCase
+{
+    /** @var Query */
+    protected $query;
+
+    public function setUp()
+    {
+        /** @var EntityManagerInterface|\PHPUnit_Framework_MockObject_MockObject $entityManager */
+        $entityManager = $this->createMock('Doctrine\ORM\EntityManagerInterface');
+        $entityManager->method('getConfiguration')->willReturn(new Configuration());
+        $this->query = new Query($entityManager);
+    }
+
+    /**
+     * @dataProvider provideGetValue
+     */
+    public function testGetValue($value, $expected, $dateFormat = 'r')
+    {
+        $iterator = new DoctrineORMQuerySourceIterator($this->query, [], $dateFormat);
+        $this->assertEquals($expected, $iterator->getValue($value));
+    }
+
+    public function provideGetValue() {
+        $nowDateTime = new \DateTime();
+
+        $dataProvider = [
+            [[1, 2, 3], null],
+            [new \ArrayIterator([1, 2, 3]), null],
+            [$nowDateTime, $nowDateTime->format('r')],
+            [$nowDateTime, $nowDateTime->format('Y-m-d H:i:s'), 'Y-m-d H:i:s'],
+            [123, 123],
+            ["123", "123"],
+            [new ObjectWithToString('object with to string'), 'object with to string'],
+        ];
+
+        if (\class_exists('\DateTimeImmutable')) {
+            $nowDateTimeImmutable = new \DateTimeImmutable();
+            $dataProvider[] = [$nowDateTimeImmutable, $nowDateTimeImmutable->format('r')];
+            $dataProvider[] = [$nowDateTimeImmutable, $nowDateTimeImmutable->format('Y-m-d H:i:s'), 'Y-m-d H:i:s'];
+        }
+
+        return $dataProvider;
+    }
+}
```

Comment:
Please configure your IDE to always add a newline at end of file, this will make the red pictogram disappear.